sharktank/sharktank/layers/configs/llm_configs.py

```diff
@@ -45,36 +46,119 @@ class LlamaHParams:
     attn_head_dim: int
     attention_layer_norm_rms_epsilon: float
     attention_head_count_kv: int
+
+    # Deepseek Multi-Latent Attention config
+    q_lora_rank: Optional[int] = None
+    kv_lora_rank: Optional[int] = None
+    qk_nope_head_dim: Optional[int] = None
+    qk_rope_head_dim: Optional[int] = None
+    v_head_dim: Optional[int] = None
+
+    # Grok Attention config
+    attention_softcap: Optional[float] = None
+
+    # RoPE config
     rope_dimension_count: Optional[int] = None
     rope_freq_base: Optional[float] = None
+
+    # Deepseek RoPE+YaRN config
+    rope_scaling_type: Optional[str] = None
+    rope_scaling_factor: Optional[float] = None
+    rope_scaling_original_context_length: Optional[int] = None
+    rope_scaling_yarn_log_multiplier: Optional[float] = None
+
+    # MoE config
     expert_count: Optional[int] = None
     expert_used_count: Optional[int] = None
 
-    # Latent Attention Config - Deepseek specific
-    nope_dim: Optional[int] = None
-    kv_latent_dim: Optional[int] = None
-    v_head_dim: Optional[int] = None
-
-    # Grok configurations
-    attention_softcap: Optional[float] = None
+    # Deepseek MoE config
+    expert_shared_count: Optional[int] = None
+    moe_intermediate_size: Optional[int] = None
+    """Size of the MoE experts feed forward network hidden dimension."""
+    n_expert_groups: Optional[int] = None
+    n_limited_groups: Optional[int] = None
+    n_dense_layers: Optional[int] = None
+    route_scale: Optional[float] = None
 
     # Deepseek MoE config
     expert_shared_count: Optional[int] = None
 
     @staticmethod
     def from_gguf_props(p: dict[str, Any]):
         name_prefix = p.get("general.architecture", "llama")
-        default_expert_count = 0
-        default_expert_used_count = 0
+
         default_rope_freq_base = 500000.0
         default_rope_dimension_count = 128
+        default_expert_count = 0
+        default_expert_used_count = 0
+        defaut_n_dense_layers = 0
+
         attention_head_count = _int_prop(p, f"{name_prefix}.attention.head_count")
         rope_dimension_count = _optional_int_prop(
             p, f"{name_prefix}.rope.dimension_count", default_rope_dimension_count
         )
+        expert_count = _optional_int_prop(
+            p, f"{name_prefix}.expert_count", default_expert_count
+        )
 
         attention_softcap = 30.0 if name_prefix == "grok" else None
 
+        if name_prefix == "deepseek2":
+            default_n_expert_groups = 8
```

Comment:
Move this to a helper to avoid the single large block for deepseek. Just makes it easier for skimming. And have the helper return a `dict`. Comment below will help that make sense.